apps/openmw/mwmechanics/aipackage.cpp

```diff
@@ -22,6 +22,27 @@
 
 #include <osg/Quat>
 
+namespace MWMechanics
+{
+    static float distance(const osg::Vec2f& lhs, const osg::Vec2f& rhs)
+    {
+        return (lhs - rhs).length2();
+    }
+
+    static float distanceIgnoreZ(const osg::Vec3f& lhs, const osg::Vec3f& rhs)
+    {
+        return distance(osg::Vec2f(lhs.x(), lhs.y()), osg::Vec2f(rhs.x(), rhs.y()));
+    }
+
+    static float distance(const osg::Vec3f& lhs, const osg::Vec3f& rhs, const MWWorld::Ptr& actor)
+    {
+        const auto world = MWBase::Environment::get().getWorld();
+        if (world->isSwimming(actor) || world->isFlying(actor))
+            return distance(lhs, rhs);
+        return distanceIgnoreZ(lhs, rhs);
+    }
```

Comment:
Maybe move these into pathfinding.hpp?

Reply:
I thought about this. Wasn't sure what's better. Decided to have this code where it's only required. Maybe we should have some separate header for this stuff and other one about distances.

Reply:
We may need these overloads for something else, keeping them in one place would be more consistent.

e.g. the overload that takes a Ptr can be used in getFightDistanceBias().

Reply:
Done